pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -438,6 +453,45 @@ func (r *ReconcileComplianceScan) phaseDoneHandler(instance *compv1alpha1.Compli
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileComplianceScan) scanDeleteHandler(instance *compv1alpha1.ComplianceScan, logger logr.Logger) (reconcile.Result, error) {
+	if common.ContainsFinalizer(instance.ObjectMeta.Finalizers, compv1alpha1.ScanFinalizer) {
+		scanToBeDeleted := instance.DeepCopy()
+
+		// Force remove rescan annotation since we're deleting the scan
+		if scanToBeDeleted.NeedsRescan() {
+			delete(scanToBeDeleted.Annotations, compv1alpha1.ComplianceScanRescanAnnotation)
+		}
+		// Force debug to be false so we actually remove dependent objects
```

Comment:
Why does debug stop object removal?

Reply:
because we need it for CI, to be able to fetch the pods' logs. This would be useful for debugging and testing in a deployment for the same reasoning. This is not desirable if you're actually deleting an object; then you REALLY want to delete everything.

Reply:
Makes sense, adding some more of this context to the comment would be helpful

Reply:
Done.